pkg/apis/postgres-operator.crunchydata.com/validation.md

```diff
@@ -0,0 +1,94 @@
+<!--
+# Copyright 2025 Crunchy Data Solutions, Inc.
+#
+# SPDX-License-Identifier: Apache-2.0
+-->
+
+# Custom Resource Definitions
+
+These directories contain Go types that serve as [DTO]s for communicating with the Kubernetes API.
+We use the [controller-gen] tool to produce [CRD]s with schemas that match the Go types.
+The CRD schema tells Kubernetes what fields and values are allowed in our API objects and how to handle changes to values.
+
+> [!TIP]
+> The CRD schema is *most* of the UX of our API
+
+CRD schemas are modified OpenAPI 3.0 [validation] schemas.
+Much of the schema defines what fields, types, and values are *allowed*.
+`controller-gen` considers the [Go type] of a field and its [validation markers] for this.
+
+Kubernetes uses its own algorithm to consider and accept changes to API objects: [Server-Side Apply], SSA.
+CRD schemas contain non-standard attributes that affect SSA.
+`controller-gen` considers the [processing markers] of a field for this.
+
+🤔 Validation is *what* and SSA/processing is *how* 🤔
+
+[controller-gen]: https://book.kubebuilder.io/reference/controller-gen
+[CRD]: https://docs.k8s.io/tasks/extend-kubernetes/custom-resources/custom-resource-definitions
+[DTO]: https://martinfowler.com/eaaCatalog/dataTransferObject.html
+[Go type]: https://go.dev/ref/spec#Types
+[Kubernetes API]: https://docs.k8s.io/concepts/overview/kubernetes-api
+[processing markers]: https://book.kubebuilder.io/reference/markers/crd-processing
+[Server-Side Apply]: https://docs.k8s.io/reference/using-api/server-side-apply
+[validation]: https://docs.k8s.io/tasks/extend-kubernetes/custom-resources/custom-resource-definitions#validation
+[validation markers]: https://book.kubebuilder.io/reference/markers/crd-validation
+
+
+# OpenAPI Properties
+
+Kubernetes supports a subset of the OpenAPI schema definition, which is a subset of an old draft of [JSON schema].
+
+Fields can have these properties:
```

Comment:
🔧 Add `nullable` around here; link to https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions#defaulting-and-nullable

Reply:
fixed up